src/libraries/System.Text.RegularExpressions/tests/Regex.Groups.Tests.cs

```diff
@@ -716,6 +716,8 @@ public static IEnumerable<object[]> Groups_Basic_TestData()
             yield return new object[] { null, @"(?:abcd|efg[hij]*)k", "efgjk", RegexOptions.None, new string[] { "efgjk" } };
             yield return new object[] { null, @"[ace]?b[ace]?b[ace]?b[ace]?b", "cbbabeb", RegexOptions.None, new string[] { "cbbabeb" } };
             yield return new object[] { null, @"[ace]?b[ace]?b[ace]?b[ace]?b", "cBbAbEb", RegexOptions.IgnoreCase, new string[] { "cBbAbEb" } };
+            yield return new object[] { null, @"a[^wz]*w", "abcdcdcdwz", RegexOptions.None, new string[] { "abcdcdcdw" } };
+            yield return new object[] { null, @"a[^wyz]*w", "abcdcdcdwz", RegexOptions.None, new string[] { "abcdcdcdw" } };
```

Comment:
Are there any tests for this same regex that don't match? For example, the string `abczw`.

Reply:
For this exact expression, no, I can add one. It's quite likely we have one covering the general case, but a little such duplication is fine.